### DISCUSSION TITLE
Unzip a Payload file in Burn Bootstrapper using WiX4

### DISCUSSION THREAD

**Discussion by Frittions** [wixtoolset/issues#7847] [Questions] [answered]
Hello,

I am building a burn bootstrapper to install the EdiMax 7822UTC driver, which comes as an InstallShield Setup.exe file. The driver file can be found [here](https://www.edimax.com/edimax/download/download/data/edimax/global/download/for_home/wireless_adapters/wireless_adapters_ac1200_dual-band/ew-7822utc), but the contents of the driver package look like this:

![image](https://github.com/wixtoolset/issues/assets/97322291/574a30b1-0ccc-4019-90dc-9fec34ad330e)

 The contents of each of those folders cantains many files (~100 files). I have got the bootstrapper to correctly install the driver, but I had to manually input every file from the driver into a Payload element. This took forever and looks really sloppy. I was wondering if there was a way to zip the contents of the driver up and pass only the zip file as a Payload element? If so, how do you unzip the file automatically when when running the installer?

Example code:

`<ExePackage Id="EdiMaxInstall" SourceFile="C:\temp\EdiMaxInstallFolder\Setup.exe" Compressed="true">`
`<Payload Name=".\EAP\EAP-FAST.msi" SourceFile="C:\temp\EdiMaxInstallFolder\EAP\EAP-FAST" />`
`<Payload Name=".\EAP\EAP-LEAP.msi" SourceFile="C:\temp\EdiMaxInstallFolder\EAP\EAP-LEAP" />`
`Payload Name=".\EAP\EAP-PEAP.msi" SourceFile="C:\temp\EdiMaxInstallFolder\EAP\EAP-PEAP" />`
`... (ADD 100 MORE FILES LIKE THIS)`
`</ExePackage>`

**Comment by wmanning** [comment]
I do use burn and I don't think you can but I'm not 100% certain.

Is there an ini file with the driver files?  One way that might work is to put the files in an MSI and use [difxapp](https://wixtoolset.org/docs/schema/difxapp/driver/).   The driver element goes in the component with the ini file. But I only install USB drivers and I don't know if it would work with other driver types and it bypasses their installer. Also I believe difxapp is depreciated and will be removed in WiX 5.

**Reply by Frittions** [reply]
No, unfortunately, there is no .ini file.

Thanks for the help though!

**Comment by robmen** [comment, marked as answer]
It's brand new, but the [HeatWave Build Tools](https://www.firegiant.com/products/heatwave-build-tools/) have a `HarvestPayloadFolder` element that can simplify that sort of scenario in .wxs code. Using a zip file for this scenario doesn't make sense because the files are already compressed in the Bundle, zipping would add more overhead for no value.
